openmc/deplete/d1s.py

```diff
@@ -0,0 +1,246 @@
+
+    Returns
+    -------
+    openmc.Tally
+        Derived tally with time correction factors applied
+
+    """
+    # Make sure the tally contains a ParentNuclideFilter
+    for i_filter, filter in enumerate(tally.filters):
+        if isinstance(filter, openmc.ParentNuclideFilter):
+            break
+    else:
+        raise ValueError('Tally must contain a ParentNuclideFilter')
+
+    # Get list of radionuclides based on tally filter
+    radionuclides = [str(x) for x in tally.filters[i_filter].bins]
+    tcf = np.array([time_correction_factors[x][index] for x in radionuclides])
+
+    # Create copy of tally
+    new_tally = deepcopy(tally)
+
+    # Determine number of bins in other filters
+    n_bins_before = prod([f.num_bins for f in tally.filters[:i_filter]])
+    n_bins_after = prod([f.num_bins for f in tally.filters[i_filter + 1:]])
+
+    # Reshape sum and sum_sq, apply TCF, and sum along that axis
+    _, n_nuclides, n_scores = new_tally.shape
+    n_radionuclides = len(radionuclides)
+    shape = (n_bins_before, n_radionuclides, n_bins_after, n_nuclides, n_scores)
+    tally_sum = new_tally.sum.reshape(shape)
+    tally_sum_sq = new_tally.sum_sq.reshape(shape)
+
+    # Apply TCF, broadcasting to the correct dimensions
+    tcf.shape = (1, -1, 1, 1, 1)
+    new_tally._sum = tally_sum * tcf
+    new_tally._sum_sq = tally_sum_sq * (tcf*tcf)
+    new_tally._mean = None
+    new_tally._std_dev = None
+
+    shape = (-1, n_nuclides, n_scores)
+
+    if sum_nuclides:
+        # Query the mean and standard deviation
+        mean = new_tally.mean
+        std_dev = new_tally.std_dev
+
+        # Sum over parent nuclides (note that when combining different bins for
+        # parent nuclide, we can't work directly on sum_sq)
+        new_tally._mean = mean.sum(axis=1).reshape(shape)
+        new_tally._std_dev = np.linalg.norm(std_dev, axis=1).reshape(shape)
+        new_tally._derived = True
+
+        # Remove ParentNuclideFilter
+        new_tally.filters.pop(i_filter)
+    else:
+        new_tally._sum.shape = shape
+        new_tally._sum_sq.shape = shape
+
+    return new_tally
+
+
+def prepare_tallies(
+        model: list[openmc.Tally],
+        nuclides: list[str] | None = None,
+        chain_file: str | None = None
+) -> list[str]:
+    """Prepare tallies for the D1S method.
+
+    This function adds a :class:`~openmc.ParentNuclideFilter` to any tally that
+    has a particle filter with a single 'photon' bin.
+
+    Parameters
+    ----------
+    model : openmc.Model
+        Model to prepare tallies for
+    nuclides : list of str, optional
+        Nuclides to use for the parent nuclide filter. If None, radionuclides
+        are determined from :func:`get_radionuclides`.
+    chain_file : str, optional
+        Chain file to use for inspecting decay data. If None, defaults to
+        ``openmc.config['chain_file']``
+
+    Returns
+    -------
+    list of str
+        List of possible radionuclides
```

Comment:
maybe something slightly more descriptive here like "List of parent nuclides being filtered on" or something along those lines.